src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Number.cs

```diff
@@ -128,7 +171,7 @@ private static bool TryParseNumber(ReadOnlySpan<byte> source, ref NumberBuffer n
                     }
                 }
 
-                int numDigitsAfterDecimalToCopy = Math.Min(srcIndex - startIndexOfDigitsAfterDecimalToCopy, NumberBuffer.BufferSize - dstIndex - 1);
+                int numDigitsAfterDecimalToCopy = Math.Min(srcIndex - startIndexOfDigitsAfterDecimalToCopy, number.Digits.Length - 1);
```

Comment:
need to fix this, it should be `number.Digits.Length - dstIndex - 1` (which can be `maxDigitCount - dstIndex`)